tests/pdctl/scheduler/scheduler_test.go

```diff
@@ -124,6 +188,114 @@ func (s *schedulerTestSuite) TestScheduler(c *C) {
+		"evict-leader-scheduler":       true,
+	}
+	for _, scheduler := range schedulers {
+		c.Assert(expected[scheduler], Equals, true)
+	}
+
+	// scheduler delete command
+	args = []string{"-u", pdAddr, "scheduler", "remove", "evict-leader-scheduler"}
+	_, _, err = pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	args = []string{"-u", pdAddr, "scheduler", "show"}
+	_, output, err = pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	schedulers = schedulers[:0]
+	c.Assert(json.Unmarshal(output, &schedulers), IsNil)
+	expected = map[string]bool{
+		"balance-leader-scheduler":     true,
+		"balance-hot-region-scheduler": true,
+		"label-scheduler":              true,
+		"grant-leader-scheduler-1":     true,
+	}
+	for _, scheduler := range schedulers {
+		c.Assert(expected[scheduler], Equals, true)
+	}
+
+	//check the compactily
+	// scheduler add command
+	args = []string{"-u", pdAddr, "scheduler", "add", "evict-leader-scheduler", "2"}
+	_, _, err = pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	args = []string{"-u", pdAddr, "scheduler", "show"}
+	_, output, err = pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	schedulers = schedulers[:0]
+	c.Assert(json.Unmarshal(output, &schedulers), IsNil)
+	expected = map[string]bool{
+		"balance-leader-scheduler":     true,
+		"balance-hot-region-scheduler": true,
+		"label-scheduler":              true,
+		"grant-leader-scheduler-1":     true,
+		"evict-leader-scheduler":       true,
+	}
+	for _, scheduler := range schedulers {
+		c.Assert(expected[scheduler], Equals, true)
+	}
+
+	// scheduler add command twice
+	args = []string{"-u", pdAddr, "scheduler", "add", "evict-leader-scheduler", "4"}
+	_, _, err = pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	args = []string{"-u", pdAddr, "scheduler", "show"}
+	_, output, err = pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	schedulers = schedulers[:0]
+	c.Assert(json.Unmarshal(output, &schedulers), IsNil)
+	expected = map[string]bool{
+		"balance-leader-scheduler":     true,
+		"balance-hot-region-scheduler": true,
+		"label-scheduler":              true,
+		"grant-leader-scheduler-1":     true,
+		"evict-leader-scheduler":       true,
+	}
+	for _, scheduler := range schedulers {
+		c.Assert(expected[scheduler], Equals, true)
+	}
+
+	//check add success
+	args = []string{"-u", pdAddr, "scheduler", "config", "show", "evict-leader-scheduler"}
+	_, output, err = pdctl.ExecuteCommandC(cmd, args...)
+	c.Assert(err, IsNil)
+	configInfo = make(map[string]interface{})
+	c.Assert(json.Unmarshal(output, &configInfo), IsNil)
+	expectedConfig["store-id-ranges"] = map[string]interface{}{"2": []interface{}{map[string]interface{}{"end-key": "", "start-key": ""}}, "4": []interface{}{map[string]interface{}{"end-key": "", "start-key": ""}}}
+	c.Assert(expectedConfig, DeepEquals, configInfo)
+
+	// evict-scheduler remove command
+	args = []string{"-u", pdAddr, "scheduler", "remove", "evict-leader-scheduler-2"}
```

Comment:
> > @nolouch about this check https://github.com/pingcap/tidb-ansible/blob/c4b9e53906e7e82eee4a59606c6f2bb31b6bc36a/common_tasks/add_evict_leader_scheduler.yml#L3-L10, when calling "scheduler remove evict-leader-scheduler-{store_id}", is ok to just throw "scheduler not found"? Because when call "scheduler show" will only show "evict-leader-scheduler".
> > And if you want to support call "scheduler add evict-leader-scheduler {store_id}" multiply times to achieve evict leader from multiply stores, or make evict-leader-scheduler not evict leader at a store by calling "scheduler remove evict-leader-scheduler-{store_id}", I think the logic should not be implemented at the http.Handler at evictSchedulerConfig but to add some check at the `server/handler.go/[addScheduler()/removeScheduler()]`.
> 
> * **`scheduler not found` is not expected.**  We need to ensure that the behavior of ansible is consistent with the previous, people may upgrade TiDB but do not upgrade ansible in some times.
> * An easy way is let the API `http://{{ pd_addr }}/pd/api/v1/schedulers/evict-leader-scheduler-{{ store_id }}` same as `http://{{ pd_addr }}/pd/api/v1/schedulers/evict-leader-scheduler`, directly remove the evict leader scheduler.

@rleungx 
because the evict-leader-scheduler config only support to add store to evict-leader-scheduler, so `remove evict-leader-scheduler-2` will remove the whole `evict-leader-scheduler` even if there are other stores or even store 2 is not in this `evict-leader-scheduler`, maybe we can add new config flag to support remove specific store from `evict-leader-scheduler`. @nolouch

Reply:
Do you mean to add a new command in `pd-ctl`?

Reply:
@nolouch maybe need add a method to handler delete like `router.HandleFunc("/delete", h.DeleteConfig).Methods("DELETE")`.

Reply:
@qiffang Good point, it is more convenient to delete a store than to update the config with a list of stores, and we should consider the situation there is only one store id in the config.

Reply:
Would you add this method as you mentioned?